include/yaml-cpp/fp_to_string.h

```diff
@@ -0,0 +1,206 @@
+// SPDX-FileCopyrightText: 2024 Simon Gene Gottlieb
+// SPDX-License-Identifier: MIT
+
+#ifndef YAML_H_FP_TO_STRING
+#define YAML_H_FP_TO_STRING
+
+#include "contrib/dragonbox.h"
+
+#include <array>
+#include <cassert>
+#include <cmath>
+#include <sstream>
+#include <tuple>
+
+namespace YAML {
+namespace detail {
+namespace fp_formatting {
+
+/**
+ * Converts a integer into its ASCII digits.
+ *
+ * @param begin/end - a buffer, must be at least 20bytes long
+ * @param value     - input value
+ * @param width     - minimum number of digits, fill with '0' to the left. Must be equal or smaller than the buffer size.
+ * @return          - number of digits filled into the buffer.
+ *
+ * Example:
+ * std::array<char, 20> buffer;
+ * auto ct = convertToChars(buffer.begin(), buffer.end(), 23, 3);
+ * assert(ct = 3);
+ * assert(buffer[0] == '0');
+ * assert(buffer[1] == '2');
+ * assert(buffer[2] == '3');
+ */
+inline auto convertToChars(char* begin, char* end, size_t value, int width=1) -> int {
+    assert(width >= 1);
+    assert(end >= begin);       // end must be after begin
+    assert(end-begin >= width); // Buffer must be large enough
+    assert(end-begin >= 20);    // 2^64 has 20digits, so at least 20 digits must be available
+
+    // count number of digits, and fill digits array accordingly
+    int digits_ct{};
+    while (value > 0) {
+        char c = value % 10 + '0';
+        value = value / 10;
+        digits_ct += 1;
+        *(end-digits_ct) = c;
+    }
+    while(digits_ct < width) {
+        assert(digits_ct < 64);
+        digits_ct += 1;
+        *(end-digits_ct) = '0';
+    }
+    // move data to the front of the array
+    std::memmove(begin, end-digits_ct, digits_ct);
+    return digits_ct;
+}
+
+/**
+ * Converts a float or double to a string.
+ *
+ * converts a value 'v' to a string. Uses dragonbox for formatting.
+ */
+template <typename T>
+auto fp_to_string(T v, int precision = 0) -> std::string {
```

Comment:
Should we name this `FPToString` instead since the rest of the codebase is using PascalCase? Otherwise, we will end up with a mix of `snake_case`, `camelCase`, and `PascalCase`.